### ISSUE TITLE
Finish removing pickle parameters from existing tasks

### ISSUE THREAD

**Issue by markstory** [getsentry/sentry#81913]
Before we can move tasks from celery to taskworker they need to not rely on pickle for parameter serialization. There are still a handful of tasks that require pickle for parameters. The tasks in `LEGACY_PICKLE_TASKS` should be refactored so that they are JSON safe.

- [ ] Update tasks in `LEGACY_PICKLE_TASKS` to have JSON safe parameters. This will take  two deploys for each task to preserve backwards compatibility.
- [ ] Enable CELERY_COMPLAIN_ABOUT_BAD_USE_OF_PICKLE in s4s to further validate that no celery tasks are using pickle parameters.

## Migrating tasks

- First add new parameters that don't require pickle.
- Deploy new task signatures (but don't call them yet)
- Do another change to update delay() and  apply_async() to use the new parameters, and stop sending the pickle params. This is a good time to update LEGACY_PICKLE_TASKS
